homeassistant/components/sharkiq/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Config flow for Shark IQ integration."""
+
+import asyncio
+from typing import Dict, Optional
+
+import aiohttp
+import async_timeout
+from sharkiqpy import SharkIqAuthError, get_ayla_api
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN, LOGGER  # pylint:disable=unused-import
+
+SHARKIQ_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+    ayla_api = get_ayla_api(
+        username=data[CONF_USERNAME],
+        password=data[CONF_PASSWORD],
+        websession=hass.helpers.aiohttp_client.async_get_clientsession(hass),
+    )
+
+    try:
+        with async_timeout.timeout(10):
+            LOGGER.debug("Initialize connection to Ayla networks API")
+            await ayla_api.async_sign_in()
+    except (asyncio.TimeoutError, aiohttp.ClientError):
+        raise CannotConnect
+    except SharkIqAuthError:
+        raise InvalidAuth
+
+    # Return info that you want to store in the config entry.
+    return {"title": f"Shark IQ ({data[CONF_USERNAME]:s})"}
```

Comment:
No need to include `Shark IQ` in the title, the name of the integration is always prefixed. Just username will do.
```suggestion
    return {"title": data[CONF_USERNAME]}
```